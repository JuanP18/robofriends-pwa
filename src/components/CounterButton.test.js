import React from 'react';
import {shallow} from 'enzyme';
import CounterButton from './CounterButton'; 

describe('Counter Button Component', () => {
  it('expect to render CounterButton component' , () => {
      const mockColor = 'red';
      expect(shallow(<CounterButton color={mockColor}/>)).toMatchSnapshot();
  });

  it('correctly increments the counter' , () => {
      const mockColor = 'red';
      const wrapper = shallow(<CounterButton color={mockColor}/>);
      
      wrapper.find('[id="counter"]').simulate('click');
      wrapper.find('[id="counter"]').simulate('click');
      expect(wrapper.state()).toEqual({count:3});
      wrapper.find('[id="counter"]').simulate('click');
      expect(wrapper.state()).toEqual({count:4});
      wrapper.find('[id="counter"]').simulate('keypress');
      expect(wrapper.state()).toEqual({count:4});
      expect(wrapper.props().color).toEqual('red');
  });

  it('the counter is not updated' , () => {
    const mockColor = 'red';
    const wrapper = shallow(<CounterButton color={mockColor}/>);
    const shouldUpdate = wrapper.instance().shouldComponentUpdate({ count: 1}, { count: 1});
    expect(shouldUpdate).toEqual(false);
});
});