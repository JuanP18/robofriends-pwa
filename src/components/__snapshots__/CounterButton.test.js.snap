// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Button Component expect to render CounterButton component 1`] = `
<button
  color="red"
  id="counter"
  onClick={[Function]}
>
  Count: 
  1
</button>
`;
